daemon/guestos_mgr.c

```diff
@@ -237,6 +218,24 @@ guestos_mgr_delete(guestos_t *os)
 	return 0;
 }
 
+void
+guestos_mgr_purge_obsolete(void)
+{
+	INFO("Looking for obsolete GuestOSes to purge...");
+	for (list_t *l = guestos_list; l;) {
+		list_t *next = l->next;
+		guestos_t *os = l->data;
+		guestos_t *latest = guestos_mgr_get_latest_by_name(guestos_get_name(os), true);
+		if (latest && guestos_get_version(os) < guestos_get_version(latest) &&
+		    guestos_mgr_is_guestos_used_by_containers(guestos_get_name(os))) {
```

Comment:
should this be "!guestos_mgr_is_guestos_used_by_containers(guestos_get_name(os)" ?

Reply:
true, will change that